packages/core/src/App/Components/Elements/Modals/RedirectNotice/redirect-notice-modal.jsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { Dialog } from '@deriv/components';
+import { localize } from '@deriv/translations';
+import { getCurrentBinaryDomain } from '@deriv/shared';
+
+const RedirectNoticeModal = ({ is_logged_in }) => {
+    const [dialog_status, setDialogStatus] = React.useState(false);
+    const [external_link, setExternalLink] = React.useState('');
+
+    const isThirdPartyLink = href => {
+        let destination;
+        try {
+            destination = new URL(href);
+        } catch (e) {
+            return false;
+        }
+        return (
+            !!destination.host &&
+            !new RegExp(`^.*\\.${getCurrentBinaryDomain() || 'binary\\.com'}$`).test(destination.host) && // destination host is not binary subdomain
+            !new RegExp('^.*\\.binary\\.bot$').test(destination.host) && // destination host is not binary subdomain
+            !/www.(betonmarkets|xodds).com/.test(destination.host) && // destination host is not binary old domain
+            !/deriv.(app|com)/.test(destination.host) && // destination host is not deriv
+            window.location.host !== destination.host
+        );
+    };
+    const onCancelDialog = () => {
+        setDialogStatus(false);
+    };
+    const onConfirmDialog = () => {
+        setDialogStatus(false);
+        window.open(external_link);
+    };
+
+    React.useEffect(() => {
+        document.addEventListener('click', function(e) {
+            if (isThirdPartyLink(e.target.href) && is_logged_in) {
+                setExternalLink(e.target.href);
+                e.preventDefault();
+                setDialogStatus(true);
+            }
+        });
```

Comment:
`click` event listener should be removed when this component is going to unmount itself.